packages/cli-core/src/package-manager-controller/package_manager_controller_base.ts

```diff
@@ -112,6 +117,29 @@ Get started by running \`${this.binaryRunner} amplify sandbox\`.`;
     );
   }
 
+  /**
+   * runWithPackageManager - Factory function that runs a command with the specified package manager's binary runner
+   */
+  runWithPackageManager(
+    args: string[] = [],
+    dir: string,
+    options?: {
+      env?: Record<string, string>;
+      stdin?: 'inherit' | 'pipe' | 'ignore';
+      stdout?: 'inherit' | 'pipe' | 'ignore';
+      stderr?: 'inherit' | 'pipe' | 'ignore';
+      extendEnv?: boolean;
+    }
```

Comment:
how many of these options do callers of `runWithPackageManager` actually use? Should we expose a more limited set of props here like `pipeOutput: true` for example?

Reply:
Done 47b615078c